jennchenn/uwexplore
# Populate DB with courses/schedule data

## Implementation description

- setup script to fetch course/schedule info from open data and save to our database in current model format
- limited course fetch query to only return 30 results
- added basic API client to frontend to help facilitate grabbing actual courses!
- completes BREAD-102



## Steps to test

1. (please do not run the seed script it takes forever and running it again will duplicate the section data ;_;)
2. Try out GET courses and see what it looks like, e.g.:
```
curl localhost:5000/courses
curl "localhost:5000/courses?code=1"
curl "localhost:5000/courses?query=syde"
```



## What should reviewers focus on?

- does APIClient make sense?
- currently only the _current_ term's schedule is saved under each course object
- ~~where will prereq/antireq info come from?~~

TODO BEFORE MERGING
- [ ] add prereq/antireq parsing
- [ ] add check for start/end times of form 01-01-01:00:00:00 => online (these will be set to `None`)
- [ ] verify what "E" day pattern means
